tests/gentropy/conftest.py

```diff
@@ -647,3 +648,46 @@ def sample_data_for_susie_inf() -> list[np.ndarray]:
     lbf_moments = np.loadtxt("tests/gentropy/data_samples/01_test_lbf_moments.csv")
     lbf_mle = np.loadtxt("tests/gentropy/data_samples/01_test_lbf_mle.csv")
     return [ld, z, lbf_moments, lbf_mle]
+
+
+@pytest.fixture()
+def sample_data_for_coloc(spark: SparkSession) -> list[Any]:
+    """Sample data for Coloc tests."""
+    overlap_df = spark.read.parquet(
+        "tests/gentropy/data_samples/coloc_test_data.snappy.parquet"
```

Comment:
How was this file generated? For semantic tests, it's easier to understand if you create a data subset in the testing module directly.
Instead of reading a file of 500 rows, create a dataframe with 2 overlapping variants, for example.

The same testing function can be parametrised for both scenarios: associations that overlap on multiple SNPs, and on a single SNP.

Reply:
It was directly extracted from the test dataset from the R package